addons/account_edi/models/account_move.py

```diff
@@ -216,37 +216,16 @@ def _get_edi_attachment(self, edi_format):
     # Import Electronic Document
     ####################################################
 
-    @api.returns('mail.message', lambda value: value.id)
-    def message_post(self, **kwargs):
+    def _get_create_invoice_from_attachment_decoders(self):
         # OVERRIDE
-        # When posting a message, analyse the attachment to check if it is an EDI document and update the invoice
-        # with the imported data.
-        res = super().message_post(**kwargs)
-
-        if len(self) != 1 or self.env.context.get('no_new_invoice') or not self.is_invoice(include_receipts=True):
-            return res
-
-        attachments = self.env['ir.attachment'].browse(kwargs.get('attachment_ids', []))
-        odoobot = self.env.ref('base.partner_root')
-        if attachments and self.state != 'draft':
-            self.message_post(body='The invoice is not a draft, it was not updated from the attachment.',
-                              message_type='comment',
-                              subtype_xmlid='mail.mt_note',
-                              author_id=odoobot.id)
-            return res
-        if attachments and self.line_ids:
-            self.message_post(body='The invoice already contains lines, it was not updated from the attachment.',
-                              message_type='comment',
-                              subtype_xmlid='mail.mt_note',
-                              author_id=odoobot.id)
-            return res
-
-        edi_formats = self.env['account.edi.format'].search([])
-        for attachment in attachments:
-            invoice = edi_formats._update_invoice_from_attachment(attachment, self)
-            if invoice:
-                break
+        res = super()._get_create_invoice_from_attachment_decoders()
+        res.append((1, self.env['account.edi.format'].search([])._create_invoice_from_attachment))
+        return res
 
+    def _get_update_invoice_from_attachment_decoders(self, invoice):
+        # OVERRIDE
+        res = super()._get_update_invoice_from_attachment_decoders(invoice)
+        res.append((1, self.env['account.edi.format'].search([])._update_invoice_from_attachment))
```

Comment:
Use `10`, `20` to define the priority... probably useless but it will ease the potential overrides

Reply:
Maybe we could do both in a single method like...
```
{
  'priority': ...,
  'create': ...,
  'update': ...,
} 
```

Reply:
Ok for the priority.
I would prefer to keep things seperate. Since both methods will be called in different flows anyway, we would have to call the method twice anyway. Also, it allows to return the decoder based on conditions more easily (see account_invoice_extract).